token/program-2022-test/tests/token_group_initialize.rs

```diff
@@ -0,0 +1,271 @@
+    let mut test_context = setup(mint_keypair, &authority).await;
+    let payer_pubkey = test_context.context.lock().await.payer.pubkey();
+    let token_context = test_context.token_context.take().unwrap();
+
+    let update_authority = Pubkey::new_unique();
+    let max_size = 10;
+    let token_group = TokenGroup::new(
+        token_context.token.get_address(),
+        Some(update_authority).try_into().unwrap(),
+        max_size,
+    );
+
+    // fails without more lamports for new rent-exemption
+    let error = token_context
+        .token
+        .token_group_initialize(
+            &token_context.mint_authority.pubkey(),
+            &update_authority,
+            max_size,
+            &[&token_context.mint_authority],
+        )
+        .await
+        .unwrap_err();
+    assert_eq!(
+        error,
+        TokenClientError::Client(Box::new(TransportError::TransactionError(
+            TransactionError::InsufficientFundsForRent { account_index: 2 }
+        )))
+    );
+
+    // fail wrong signer
+    let not_mint_authority = Keypair::new();
+    let error = token_context
+        .token
+        .token_group_initialize_with_rent_transfer(
+            &payer_pubkey,
+            &not_mint_authority.pubkey(),
+            &update_authority,
+            max_size,
+            &[&not_mint_authority],
+        )
+        .await
+        .unwrap_err();
+    assert_eq!(
+        error,
+        TokenClientError::Client(Box::new(TransportError::TransactionError(
+            TransactionError::InstructionError(
+                1,
+                InstructionError::Custom(TokenGroupError::IncorrectMintAuthority as u32)
+            )
+        )))
+    );
+
+    token_context
+        .token
+        .token_group_initialize_with_rent_transfer(
+            &payer_pubkey,
+            &token_context.mint_authority.pubkey(),
+            &update_authority,
+            max_size,
+            &[&token_context.mint_authority],
+        )
+        .await
+        .unwrap();
+
+    // check that the data is correct
+    let mint_info = token_context.token.get_mint_info().await.unwrap();
+    let group_bytes = mint_info.get_extension_bytes::<TokenGroup>().unwrap();
+    let fetched_group = pod_from_bytes::<TokenGroup>(group_bytes).unwrap();
+    assert_eq!(fetched_group, &token_group);
+
+    // fail double-init
+    let error = token_context
+        .token
+        .token_group_initialize_with_rent_transfer(
+            &payer_pubkey,
+            &token_context.mint_authority.pubkey(),
+            &update_authority,
+            max_size,
+            &[&token_context.mint_authority],
+        )
+        .await
+        .unwrap_err();
+    assert_eq!(
+        error,
+        TokenClientError::Client(Box::new(TransportError::TransactionError(
+            TransactionError::InstructionError(
+                1,
+                InstructionError::Custom(TokenError::ExtensionAlreadyInitialized as u32)
+            )
+        )))
+    );
```

Comment:
This is probably failing in CI because it looks like the exact same transaction it's already seen. This is a strange case though, because it should be tacking on a transfer for a different amount :thinking: but either way, to make it more secure, just change `max_size` to some other number, and that way you're guaranteed to get a unique transaction.

Reply:
Actually the amount would be the same, since it's not checking if the extension exists in the client-side function. I'm not sure if we want to add that to the client, or just let the program handle that, but since `TokenGroup` is fixed it will always be the same rent each time that's called.

Still seems weird that it would work about 50% of the time, but maybe it's all about timing? 😅

Reply:
Ah, there's your answer then! We should probably smarten that up. And yeah, especially in CI, you'll often get the results of a previous transaction if it has the same signature as a previous one.